pkgs/servers/peertube/default.nix

```diff
@@ -0,0 +1,200 @@
+{ lib
+, callPackage
+, stdenv
+, fetchzip
+, youtube-dl
+, fetchurl
+, fetchFromGitHub
+, python
+, nodejs
+, nodePackages
+}:
+
+let
+  fetchedGithub = path:
+    let
+      json = lib.importJSON path;
+    in rec {
+      version = json.tag;
+      name = "${json.meta.name}-${version}";
+      src = fetchFromGitHub json.github;
+    };
+
+  yarn2nixPackage = let
```

Comment:
I don't think this is good. Instead the tool should be part of nixpkgs.

Another problem is: If the source is required to evaluate the derivation (not build it, just evaluate) then that means that nix needs to download the source even to just build the .drv file

An idea would be to add the yarn.lock to nixpkgs aswell as the tool and then pull the yarn.lock from nixpkgs, so evaluation does not depend on external dependecies

Reply:
Inspiration can be taken from https://github.com/mkg20001/nix-node-package/blob/master/nix/default.nix which solves that problem and also the one where the nodeHeaders are pulled separately

Reply:
@immae care to contribute this? I'm new to the node packaging infra and thus do not entirely understand how to approach this.

Reply:
There is the yarn2nix-moretea which contains exactly what we need and didn’t exist at the time when I wrote this package, I’ll try to make a use of it and give feedback (there seem to be some kind of issues with yarn packages coming from github)

Reply:
Shouldn't you be able to use `mkYarnPackage` here?